bcitflex/course.py

```diff
@@ -4,33 +4,35 @@
 from sqlalchemy import not_, select
 
 from bcitflex.app_functions import ModelFilter
-from bcitflex.db import DBSession
+from bcitflex.db import DBSession, db
 from bcitflex.model import Course, Meeting, Offering, Subject, Term
 from bcitflex.model.offering import NOT_AVAILABLE
 
 bp = Blueprint("course", __name__)
 
 
 @bp.route("/")
-@bp.route("/courses", methods=["GET", "POST"])
+@bp.route("/courses", methods=["GET"])
 def index():
     filters = ModelFilter(Course)
-    if request.method == "POST":
-        filters.where(Offering.term_id == request.form.get("term"))
-        filters.where(Course.subject_id == request.form.get("subject"))
-        filters.where(Meeting.campus == request.form.get("campus"))
-        filters.where(Course.code == request.form.get("code"))
-        if name := request.form.get("name"):
-            filters.where(Course.name.ilike("%" + name + "%"))
-        if request.form.get("available") is not None:
-            filters.where(not_(Offering.status.in_(NOT_AVAILABLE)))
-    courses = DBSession.scalars(filters.stmt).all()
+    filters.where(Offering.term_id == request.args.get("term"))
+    filters.where(Course.subject_id == request.args.get("subject"))
+    filters.where(Meeting.campus == request.args.get("campus"))
```

Comment:
Somehow after the change SQLAlchemy isn't able to join `Meeting` anymore. Could you please pass the association table to the call? 
```py
filters.where(Meeting.campus == request.args.get("campus"), links=[Offering])
```

Reply:
@jonbiemond 

sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DuplicateAlias) table name "offering" specified more than once

[SQL: SELECT DISTINCT course.course_id, course.subject_id, course.code, course.name, course.prerequisites, course.credits, course.url, course.created_at, course.updated_at, course.deleted_at
FROM course JOIN offering ON course.course_id = offering.course_id AND offering.deleted_at IS NULL JOIN offering ON course.course_id = offering.course_id AND offering.deleted_at IS NULL
WHERE (offering.status NOT IN (%(status_1_1)s, %(status_1_2)s, %(status_1_3)s, %(status_1_4)s, %(status_1_5)s)) AND course.deleted_at IS NULL
 LIMIT %(param_1)s OFFSET %(param_2)s]
[parameters: {'param_1': 25, 'param_2': 25, 'status_1_1': 'Full', 'status_1_2': 'In Progress', 'status_1_3': 'Cancelled', 'status_1_4': 'Closed', 'status_1_5': 'Waitlist'}]


did you test out your code before merging ?

Reply:
@jonbiemond  you code does not work with 2+ conditions

Reply:
@MikeVarchola I see the problem, sorry about that. I'll add a test for that condition and put out a fix by EOD.

Reply:
@jonbiemond  try filter for campus + filter for subject at same time

Reply:
@MikeVarchola I pushed a fix to dev with https://github.com/jonbiemond/BCIT-Flex/pull/76/commits/831fcddb8a4a7791e663971cf0959ed3cc9679cf .